README.md
## Twitter Online Tracker of the Chilean Referendum for a New Constitution

[![Version](https://img.shields.io/badge/version-v0.1.0-blue.svg)](https://github.com/bgonzalezbustamante/twConstitution/blob/master/changelog.txt) [![GitHub issues](https://img.shields.io/github/issues/bgonzalezbustamante/twConstitution.svg)](https://github.com/bgonzalezbustamante/twConstitution/issues/) [![GitHub issues-closed](https://img.shields.io/github/issues-closed/bgonzalezbustamante/twConstitution.svg)](https://github.com/bgonzalezbustamante/twConstitution/issues?q=is%3Aissue+is%3Aclosed) [![DOI](https://img.shields.io/badge/DOI-10.17605%2FOSF.IO%2F73NDB-blue)](http://doi.org/10.17605/OSF.IO/73NDB) [![License](https://img.shields.io/badge/license-CC--BY--4.0-black)](https://github.com/bgonzalezbustamante/twConstitution/blob/master/LICENSE.txt) [![R](https://img.shields.io/badge/made%20with-R%20v4.0.2-1f425f.svg)](https://cran.r-project.org/) [![Jekyll](https://img.shields.io/badge/made%20with-Jekyll-1f425f.svg)](https://jekyllrb.com/)

This is a Twitter online tracker of the Chilean referendum for a new Constitution in October 2020, which contains daily datasets on #Apruebo and #Rechazo opinions on social media.

- #Apruebo dataset, September 26, 2020 (n = 21,982; 23.1 MB)
- #Rechazo dataset, September 26, 2020 (n = 31,244; 33.9 MB)

This repository is linked to **OSF-Project** (DOI: [10.17605/OSF.IO/73NDB](http://doi.org/10.17605/OSF.IO/73NDB)). This site will be updated on a regular basis, see the [changelog](https://github.com/bgonzalezbustamante/twConstitution/blob/master/changelog.txt).

### About Me

I am a PRS in the DPhil in Politics programme in the [Department of Politics and International Relations](https://www.politics.ox.ac.uk/) and [St Hilda's College](https://www.sthildas.ox.ac.uk/) at the [University of Oxford](http://www.ox.ac.uk/), United Kingdom. Furthermore, I am an Instructor in the Department of Public Administration and Policy at the [Faculty of Economics and Management](https://fae.usach.cl/) at the Universidad de Santiago, where I have been working since 2014. 

Further details on my [website at the University of Oxford](http://users.ox.ac.uk/~shil5311/) or my [landing page](https://bgonzalezbustamante.com/).


CHANGELOG.md
# Changelog
All notable changes to "twConstitution" will be documented in this file.

## [Unreleased]

## [Released]

## [2.0.1] - 2023-03-03
### Changed
- Minor corrections.

## [2.0.0] - 2023-02-11
### Added
- Finishing scraping in JSON format on secured repository.
### Changed
- Badges in SVG on GitHub repository.
- Minor corrections.

## [1.6.20] - 2023-02-09
### Added
- Starting scraping in JSON format on secured repository.

## [1.5.20] - 2023-01-25
### Changed
- Minor corrections.

## [1.5.19] - 2022-09-01
### Changed
- Minor corrections.

## [1.5.18] - 2022-07-24
### Changed
- Minor corrections.

## [1.5.17] - 2022-04-05
### Changed
- Revision of upgrade versioning.
- Minor corrections.

## [1.5.16] - 2022-04-01
### Changed
- Minor corrections.

## [1.5.15] - 2022-03-31
### Changed
- Minor corrections.

## [1.5.14] - 2022-03-20
### Changed
- twConstitution raw data in CSV UTF-8 format on secured repository.

## [1.5.13] - 2022-03-19
### Added
- MIT license on GitHub repository.
### Changed
- twConstitution raw data in CSV UTF-8 format on secured repository.
- Scripts and folders names standardisation.
- Revision of upgrade versioning.
- Minor corrections.

## [1.4.13] - 2021-11-17
### Added
- Repository status badge.
### Changed
- Minor corrections.

## [1.3.13] - 2021-11-15
### Changed
- License in MD format on GitHub repository.
- Minor corrections.

## [1.3.12] - 2021-10-31
### Changed
- Minor corrections.

## [1.3.11] - 2021-10-22
### Changed
- Revision of upgrade versioning.
- Changelog in MD format on GitHub repository.

## [1.3.10] - 2021-09-24
### Added
- Citation CFF file.

## [1.2.10] - 2021-09-15
### Changed
- Revision of upgrade versioning.
- Minor corrections.

## [1.2.9] - 2021-08-18
### Changed
- Minor corrections.
### Deprecated
- Automated data scraping #Elecciones2021CL.

## [1.2.8] - 2021-05-15
### Added
- Automated data scraping #Elecciones2021CL.
### Changed
- Revision of upgrade versioning.

## [1.1.8] - 2021-04-20
### Added
- twConstitution raw data in CSV UTF-8 format on secured repository.

## [1.0.8] - 2021-03-27
### Changed
- Revision of upgrade versioning.
- Minor corrections.

## [1.0.7] - 2021-03-11
### Changed
- Minor corrections.

## [1.0.6] - 2021-01-12
### Changed
- Minor corrections.

## [1.0.5] - 2020-12-21
### Changed
- Minor corrections.

## [1.0.4] - 2020-10-27
### Changed
- Minor corrections.

## [1.0.3] - 2020-09-30
### Changed
- Structure of directories.
- Minor corrections.

## [1.0.2] - 2020-09-28
### Changed
- Minor corrections.

## [1.0.1] - 2020-09-27
### Changed
- Minor corrections.

## [1.0.0] - 2020-09-26
### Added
- GitHub repository.
- Automated data scraping #Apruebo.
- Automated data scraping #Rechazo.
- README in MD format on GitHub repository.
- OSF project (DOI: 0.17605/OSF.IO/73NDB).
- Badges in SVG on GitHub repository.
- GitHub page powered by Jekyll.
- Changelog in TXT format on GitHub repository.
- License CC BY 4.0 on GitHub repository.


Scraper-twConstitution.R
## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Script ID ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Twitter Online Tracker Scraper Code
## R version 4.0.2 (2020-06-22) -- "Taking Off Again"
## Date: September 2020 - May 2021

## Bastián González-Bustamante (University of Oxford, UK)
## E-mail: bastian.gonzalezbustamante@politics.ox.ac.uk
## https://bgonzalezbustamante.com/

## Twitter Online Tracker of the Chilean Referendum for a New Constitution
## OSF-Project DOI: 10.17605/OSF.IO/73NDB
## http://bgonzalezbustamante.github.io/twConstitution/

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Packages and Credentials

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Packages
library(rtweet)
library(dplyr)
library(tidytext)
## library(ggplot2)
## library(quanteda)

## Bastián González-Bustamante' Credentials
appname <- "API_NAME_HERE"
apikey <- "CONSUMER_KEY_HERE"
apisecret <- "CONSUMER_SECRET_HERE"
accesstoken <- "APP_TOKEN_HERE"
accesstokensecret <- "APP_SECRET_HERE"

## Twitter Token
twitter_token <- create_token(app = appname,
                              consumer_key = apikey,
                              consumer_secret = apisecret,
                              access_token = accesstoken,
                              access_secret = accesstokensecret)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Referendum 2020 ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Data Collection
## rm(apruebo_tweets)
## rm(rechazo_tweets)

apruebo_tweets <- search_tweets(q = "#Apruebo", n = 100000, lang = "es",
                                include_rts = FALSE, 
                                retryonratelimit = TRUE)

rechazo_tweets <- search_tweets(q = "#Rechazo", n = 100000, lang = "es",
                                include_rts = FALSE, 
                                retryonratelimit = TRUE)

## Export as CSV

save_as_csv(apruebo_tweets, 'dataCollection/20201101_twApruebo.csv', 
            prepend_ids = TRUE, na = "", fileEncoding = "UTF-8")

## test1 <- read.csv("dataCollection/20201101_twApruebo.csv", header = T, sep = ",")
## rm(test1)

save_as_csv(rechazo_tweets, 'dataCollection/20201101_twRechazo.csv', 
            prepend_ids = TRUE, na = "", fileEncoding = "UTF-8")

## test2 <- read.csv("dataCollection/20201101_twRechazo.csv", header = T, sep = ",")
## rm(test2)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Elections 2021 ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Data COllection
## rm(electoral_tweets)
electoral_tweets <- search_tweets(q = "#Elecciones2021CL", n = 100000, lang = "es",
                                  include_rts = FALSE, 
                                  retryonratelimit = TRUE)

## Export as CSV
save_as_csv(electoral_tweets, '../secured-data/twConstitution/electionsData/20210517_Elecciones2021CL.csv', 
            prepend_ids = TRUE, na = "", fileEncoding = "UTF-8")


code/scraper_twConstitution.R
## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Script ID ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Twitter Online Tracker Scraper Code
## R version 4.0.2 (2020-06-22) -- "Taking Off Again"
## Date: September 2020 - May 2021

## Bastián González-Bustamante (University of Oxford, UK)
## E-mail: bastian.gonzalezbustamante@politics.ox.ac.uk
## https://bgonzalezbustamante.com/

## Twitter Online Tracker of the Chilean Referendum for a New Constitution
## OSF-Project DOI: 10.17605/OSF.IO/73NDB
## http://bgonzalezbustamante.github.io/twConstitution/

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Packages and Credentials

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Packages
library(rtweet)
library(dplyr)
library(tidytext)
## library(ggplot2)
## library(quanteda)

## Bastián González-Bustamante' Credentials
appname <- "API_NAME_HERE"
apikey <- "CONSUMER_KEY_HERE"
apisecret <- "CONSUMER_SECRET_HERE"
accesstoken <- "APP_TOKEN_HERE"
accesstokensecret <- "APP_SECRET_HERE"

## Twitter Token
twitter_token <- create_token(app = appname,
                              consumer_key = apikey,
                              consumer_secret = apisecret,
                              access_token = accesstoken,
                              access_secret = accesstokensecret)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Referendum 2020 ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Data Collection
## rm(apruebo_tweets)
## rm(rechazo_tweets)

apruebo_tweets <- search_tweets(q = "#Apruebo", n = 100000, lang = "es",
                                include_rts = FALSE, 
                                retryonratelimit = TRUE)

rechazo_tweets <- search_tweets(q = "#Rechazo", n = 100000, lang = "es",
                                include_rts = FALSE, 
                                retryonratelimit = TRUE)

## Export as CSV

save_as_csv(apruebo_tweets, 'dataCollection/20201101_twApruebo.csv', 
            prepend_ids = TRUE, na = "", fileEncoding = "UTF-8")

## test1 <- read.csv("dataCollection/20201101_twApruebo.csv", header = T, sep = ",")
## rm(test1)

save_as_csv(rechazo_tweets, 'dataCollection/20201101_twRechazo.csv', 
            prepend_ids = TRUE, na = "", fileEncoding = "UTF-8")

## test2 <- read.csv("dataCollection/20201101_twRechazo.csv", header = T, sep = ",")
## rm(test2)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Elections 2021 ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Data COllection
## rm(electoral_tweets)
electoral_tweets <- search_tweets(q = "#Elecciones2021CL", n = 100000, lang = "es",
                                  include_rts = FALSE, 
                                  retryonratelimit = TRUE)

## Export as CSV
save_as_csv(electoral_tweets, '../secured-data/twConstitution/electionsData/20210517_2_Elecciones2021CL.csv', 
            prepend_ids = TRUE, na = "", fileEncoding = "UTF-8")
